sdk/core/core-lro/src/poller/poller.ts

```diff
@@ -100,25 +105,36 @@ export function buildCreatePoller<TResponse, TResult, TState extends OperationSt
     const cancelErrMsg = "Operation was canceled";
     let currentPollIntervalInMs = intervalInMs;
 
-    const poller: SimplePollerLike<TState, TResult> = {
-      getOperationState: () => state,
-      getResult: () => state.result,
-      isDone: () => ["succeeded", "failed", "canceled"].includes(state.status),
-      isStopped: () => resultPromise === undefined,
-      stopPolling: () => {
-        abortController.abort();
+    const poller: PollerLike<TState, TResult> = {
+      get operationState(): TState | undefined {
+        return state;
+      },
+      get result(): TResult | undefined {
+        return state?.result;
+      },
+      get isDone(): boolean {
+        return ["succeeded", "failed", "canceled"].includes(state?.status ?? "");
+      },
+      get isStopped(): boolean {
+        return resultPromise === undefined;
       },
-      toString: () =>
-        JSON.stringify({
-          state,
-        }),
       onProgress: (callback: (state: TState) => void) => {
         const s = Symbol();
         handlers.set(s, callback);
         return () => handlers.delete(s);
       },
-      pollUntilDone: (pollOptions?: { abortSignal?: AbortSignalLike }) =>
-        (resultPromise ??= (async () => {
+      serialize: async () => {
+        await statePromise;
```

Comment:
Can you please file an issue to make sure this is roundtrippable with other languages?

Reply:
do you mean cross-language rehydration? issue filed: https://github.com/Azure/azure-sdk-for-js/issues/28647